gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClientV1Impl.java

```diff
@@ -179,26 +153,283 @@ public KubernetesClientV1Impl(Vertx vertx, KubernetesConfig kubernetesConfig) {
                         return Maybe.error(
                             new RuntimeException(
                                 String.format(
-                                    "Unable to retrieve configmap %s from namespace %s. error code %d",
-                                    configmapName,
-                                    namespace,
+                                    "Unable to retrieve %s %s from namespace %s. error code %d",
+                                    resource.type.value,
+                                    resource.name,
+                                    resource.namespace,
                                     response.statusCode()
                                 )
                             )
                         );
                     } else {
-                        ConfigMap configMap = response.bodyAsJsonObject().mapTo(ConfigMap.class);
-                        if (configMap == null) {
-                            LOGGER.error("Unable to retrieve configmap {} from namespace {}", configmapName, namespace);
+                        T data = null;
+                        if (resource.key == null) {
+                            data = response.bodyAsJsonObject().mapTo(type);
+                        } else {
+                            Object item = response.bodyAsJsonObject().mapTo(type);
+                            if (item instanceof ConfigMap) {
+                                data = (T) ((ConfigMap) item).getData().get(resource.key);
+                            } else if (item instanceof Secret) {
+                                data = (T) ((Secret) item).getData().get(resource.key);
+                            }
```

Comment:
Not sure to understand why there is a difference of processing for `ConfigMap` and `Secret` whereas it was not the case in the previous version.

Reply:
The difference is when you want to get a specific key inside the ConfigMap or Secret, for example tls.key inside a secret or management.mongodb.dbname inside a ConfigMap. Before getting the key we need to first cost the event object.
But if they didn't specify the key and the want the whole secret or configmap, then there is no difference 
buffer.toJsonObject().mapTo(type);